tests/providers/aws/e2e_cli.py

```diff
@@ -0,0 +1,332 @@
+  @IgnoreWarnings
+  def testStartVm(self):
+    """End to end test on AWS.
+
+    Test creating an analysis VM and attaching a copied volume to it.
+    """
+
+    volume_copy = self._RunCreateVolumeCopy(
+        self.zone, volume_id=self.volume_to_copy)
+    self.volumes.append((self.aws, volume_copy))
+    # Create and start the analysis VM and attach the boot volume
+    self.analysis_vm = self._RunStartAnalysisVm(
+        self.analysis_vm_name,
+        self.zone,
+        attach_volumes=[volume_copy.volume_id]
+    )
+
+    # The forensic instance should be live in the analysis AWS account and
+    # the volume should be attached
+    instance = self.aws.ResourceApi(EC2_SERVICE).Instance(
+        self.analysis_vm.instance_id)
+    self.assertEqual(instance.instance_id, self.analysis_vm.instance_id)
+    self.assertIn(volume_copy.volume_id,
+                  [vol.volume_id for vol in instance.volumes.all()])
+
+  @typing.no_type_check
+  def _StoreVolumeForCleanup(self, aws_account, volume):
+    """Store a volume for cleanup when tests finish.
+
+    Args:
+      aws_account (AWSAccount): The AWS account to use.
+      volume (boto3.resource.volume): An AWS volume.
+    """
+    self.volumes.append((aws_account, volume))
+
+  @classmethod
+  @typing.no_type_check
+  def _RunStartAnalysisVm(cls, vm_name, zone, attach_volumes=None):
+    cmd = cls.cli.PrepareStartAnalysisVmCmd(
+        vm_name, zone, attach_volumes=attach_volumes)
+    try:
+      output = subprocess.check_output(
+          cmd.split(), stderr=subprocess.STDOUT, shell=False)
+      logger.info(output)
+    except subprocess.CalledProcessError as error:
+      raise ResourceCreationError(
+          'Failed creating VM in account profile {0:s}'.format(
+              cls.aws.aws_profile), __name__) from error
+    try:
+      return cls.aws.ec2.GetInstancesByName(vm_name)[0]
+    except ResourceNotFoundError as error:
+      raise ResourceNotFoundError(
+          'Failed finding created VM in account profile {0:s}'.format(
+              cls.aws.aws_profile), __name__) from error
+
+  @classmethod
+  @typing.no_type_check
+  def _RunCreateVolumeCopy(
+      cls, zone, dst_zone=None, instance_id=None, volume_id=None):
+    cmd = cls.cli.PrepareCreateVolumeCopyCmd(
+        zone, dst_zone=dst_zone, instance_id=instance_id, volume_id=volume_id)
+    try:
+      output = subprocess.check_output(
+          cmd.split(), stderr=subprocess.STDOUT, shell=False)
+      if not output:
+        raise ResourceCreationError(
+            'Could not find volume copy result', __name__)
+      volume_copy_name = output.decode('utf-8').split(' ')[-1]
+      volume_copy_name = volume_copy_name[:volume_copy_name.rindex('copy') + 4]
+      logger.info(output)
+      logger.info("Volume successfully copied to {0:s}".format(
+          volume_copy_name))
+    except subprocess.CalledProcessError as error:
+      raise ResourceCreationError('Failed copying volume', __name__) from error
```

Comment:
You want this to be as close to the `subprocess` call as possible (no use to have the decode / logging operations inside this try block)